backend/router/authentication.py

```diff
@@ -1,73 +1,115 @@
+        return v
 
-class LoginModel(BaseModel): # login model 
+
+class LoginModel(BaseModel):
     user_email: str
     user_password: str
 
-def hash_password(password: str) -> str: 
+
+def hash_password(password: str) -> str:
     return pwd_context.hash(password)
 
-def verify_password(plain_password: str, hashed_password: str) -> bool: 
+
+def verify_password(plain_password: str, hashed_password: str) -> bool:
     return pwd_context.verify(plain_password, hashed_password)
 
+
 @AUTH_ROUTER.post("/register")
 def register(user: UserModel, db: Session = Depends(get_db)):
-    existing_user = db.query(User).filter_by(user_email=user.user_email).first() 
-    if existing_user:
-        raise HTTPException(status_code=400, detail="User already exists")
-
-    hashed_password = hash_password(user.user_password)
-    new_user = User(
-        first_name=user.first_name,
-        last_name=user.last_name,
-        user_email=user.user_email,
-        user_password=hashed_password,
-        role=user.role,
-        registered_at=datetime.now(),
-    )
-    db.add(new_user)
-    db.commit()
-
-
-    token = create_access_token({"sub": new_user.user_email}) #tu sa predieli user token kory sa vytvori
-    return {"access_token": token, "token_type": "bearer"}
+    try:
+        existing_user = db.query(User).filter_by(user_email=user.user_email).first()
+        if existing_user:
+            logger.warning(
+                f"Registration attempt with existing email: {user.user_email}"
+            )
+            raise HTTPException(status_code=400, detail="User already exists")
+
+        hashed_password = hash_password(user.user_password)
+        new_user = User(
+            first_name=user.first_name,
+            last_name=user.last_name,
+            user_email=user.user_email,
+            user_password=hashed_password,
+            role=user.role,
+            registered_at=datetime.now(),
+        )
+        db.add(new_user)
+        db.commit()
+        logger.info(f"New user registered: {user.user_email}")
+
+        token = create_access_token({"sub": new_user.user_email})
+        return {"access_token": token, "token_type": "bearer"}
+    except Exception as e:
+        logger.error(f"Registration failed: {str(e)}")
+        raise HTTPException(status_code=400, detail=str(e))
+
 
 @AUTH_ROUTER.post("/login")
 def login(user: LoginModel, db: Session = Depends(get_db)):
-    db_user = db.query(User).filter_by(user_email=user.user_email).first() 
+    db_user = db.query(User).filter_by(user_email=user.user_email).first()
     if not db_user or not verify_password(user.user_password, db_user.user_password):
-        raise HTTPException(status_code=400, detail="Invalid email or password") 
+        logger.warning(f"Failed login attempt for {user.user_email}")
+        raise HTTPException(status_code=400, detail="Invalid email or password")
 
     token = create_access_token({"sub": db_user.user_email})
```

Comment:
Funguje to, ale obyčajne sa priamo do tokenu zakódujú dáta o používateľovi - rola, meno, atď.
potom sa nemusí ani robiť zakaždým request na `/authentication/me`, pretože stačí ak by sa na frontende dekódoval token

ale dobre, uvidíme neskôr že či to budeme meniť, pretože potom by sa musel robiť token parsing na frontende aby sme vedeli zobraziť/skryť nejaké UI prvky z informácií o tom tokene, atď.